server/src/actors/events.rs

```diff
@@ -152,12 +152,17 @@ impl EventProcessor {
             if self.config.processing_enabled() {
                 let geoip_lookup = self.geoip_lookup.as_ref().map(Arc::as_ref);
                 let auth = message.meta.auth();
+                let key_config = message
+                    .project_state
+                    .get_public_key_config(&auth.public_key());
```

Comment:
Please put the Public key config in `EventMeta`, so that it is not optional here.

Reply:
I would like to defer this refactor, but then also fold EventAction into KeyConfig